core/src/edu/cornell/gdiac/main/SpinnerModel.java

```diff
@@ -0,0 +1,102 @@
+package edu.cornell.gdiac.main;
+
+import com.badlogic.gdx.math.Vector2;
+import com.badlogic.gdx.physics.box2d.BodyDef.BodyType;
+import com.badlogic.gdx.physics.box2d.Joint;
+import com.badlogic.gdx.physics.box2d.World;
+import com.badlogic.gdx.physics.box2d.joints.RevoluteJointDef;
+import com.badlogic.gdx.utils.JsonValue;
+import edu.cornell.gdiac.assets.AssetDirectory;
+import edu.cornell.gdiac.physics.obstacle.ComplexObstacle;
+import edu.cornell.gdiac.physics.obstacle.Obstacle;
+import edu.cornell.gdiac.physics.obstacle.WheelObstacle;
+import edu.cornell.gdiac.util.TiledJsonParser;
+
+/**
+ * A BoxObstacle that represents a spinner, which can rotate around its center or a pivot point.
+ * <p>
+ * This class can be initialized from a JSON file, allowing for easy level design and modification.
+ */
+public class SpinnerModel extends ComplexObstacle {
+
+  private float defaultPivotRadius = 0.05f;
+  private float pinDensity = 10.0f;
+
+  /**
+   * The primary spinner obstacle
+   */
+  private SpinnerPlatformModel barrier;
+
+  /**
+   * The point around which the spinner rotates, relative to its center.
+   */
+  private Vector2 pivotPoint;
+
+  /**
+   * Creates a new SpinnerModel with default settings. The spinner does not rotate until its
+   * properties are set, typically by loading from a JSON file.
+   */
+  public SpinnerModel() {
+    super(100.0f, 200.0f); // Default size, can be overridden by JSON initialization
+    this.pivotPoint = new Vector2(100.0f, 200.0f); // Default pivot at the center
+  }
+
+  @Override
+  protected boolean createJoints(World world) {
+    assert bodies.size > 0;
+
+    Obstacle barrier = bodies.get(0);
+    Obstacle pin = bodies.get(1);
+
+    RevoluteJointDef jointDef = new RevoluteJointDef();
+    jointDef.bodyA = barrier.getBody();
+    jointDef.bodyB = pin.getBody();
+
+    jointDef.collideConnected = false;
+
+    Joint joint = world.createJoint(jointDef);
+    joints.add(joint);
+    return true;
+  }
+
+  /**
+   * Initializes the spinner from a JSON definition.
+   *
+   * @param json The JSON object defining the spinner.
+   */
+  public void initialize(AssetDirectory directory, JsonValue json, int tileHeight, Vector2 scale) {
+    barrier = new SpinnerPlatformModel();
+    barrier.setName(json.get("name").asString());
+    TiledJsonParser.initObjectFromJson(barrier, directory, json, scale, tileHeight);
+    float posX = json.getFloat("x"); // Adjusted for center
+    float posY = (tileHeight - json.getFloat("y")); // Adjusted for center
+    float radius = defaultPivotRadius;
+    bodies.add(barrier);
+
+    // Create the pin as a static WheelObstacle at the spinner's pivot point
+    pivotPoint.set(posX, posY);
+    WheelObstacle pin = new WheelObstacle(pivotPoint.x, pivotPoint.y, radius);
+    TiledJsonParser.initObjectFromJson(pin, directory, json, scale, tileHeight);
+    pin.setName(getName() + "_pin");
+    pin.setDensity(pinDensity);
+    pin.setBodyType(BodyType.StaticBody);
```

Comment:
I'm a bit confused about how you're using the `initObjectFromJson` method here. It seems like you proceed to manually manually initialize much of what this method does for you right after it. If it needs custom initialization code for the pivot that's fine, but then there shouldn't be a need to call this method.